### PR TITLE
docs: iteration on aws k8s upgrade docs

### PR DESCRIPTION
**Pull request by consideRatio**
- **No longer assumes empty user nodes**
  Our docs previously assumed there were no users on the user nodes, assuming a re-creation upgrade strategy could be used. This assumption is removed in this iteration.
- **Removed out of scope section**
  I removed the pre-requiesite section "Consider changes to `template.jsonnet`" that I now consider too out of scope to be suggested in docs to be done during a k8s upgrade.
- **Refactoring to centralize misc content**
  Node upgrade strategies, AWS auth, notes on version skew, and maybe something more.
- **Check cluster status and activity**
  Added step to overview what goes on in the cluster before upgrading. This can for example be used to rule out that something broke because of the upgrade (because it was already broken).
- **Facilitate upgrading multiple clusters in parallell**
  Upgrading multiple clusters in parallell is reasonable, and I've now made it so that the guide is easier to scale to run in parallell.

## Related

This was worked in preparation for #4009. If it wasn't done now, it would be harder to handle #4009 even though it wasn't part of scope of #4009 to get this done.

## Review

I think the time efficient approach is to let this be practically reviewed by merging it and then using it - iterating on it further if needed to fix issues with it.

### CODE REVIEW

File: docs/howto/upgrade-cluster/aws.md [hunk 1]
```diff
@@ -2,74 +2,17 @@
...
-   git status
+   Refer to [](new-cluster:prerequisites) on how to do this.
 
-   # restore changes to all files but the .jsonnet files
-   git add *.jsonnet
-   git checkout ..  # .. should be the git repo's root
-   git reset
-
-   # inspect changes
-   git diff
```
Comment (consideRatio) [inline docs/howto/upgrade-cluster/aws.md:66]: This is out of scope for k8s maintenance, and more in scope for whenever something changes in the template.

File: docs/howto/upgrade-cluster/aws.md [hunk 2]
```diff
@@ -2,74 +2,17 @@
 
 # Upgrade Kubernetes cluster on AWS
 
-```{warning}
-This upgrade will cause disruptions for users and trigger alerts for
-[](uptime-checks). To help other engineers, communicate that your are starting a
-cluster upgrade in the `#maintenance-notices` Slack channel and setup a [snooze](uptime-checks:snoozes)
```
Comment (consideRatio) [inline docs/howto/upgrade-cluster/aws.md:8]: We shouldn't need a snooze, so that was removed. I relocated the note on communicating in slack to a dedicated step.

File: docs/howto/upgrade-cluster/aws.md [hunk 3]
```diff
@@ -2,74 +2,17 @@
...
-```
-
-```{warning}
-We haven't yet established a policy for planning and communicating maintenance
-procedures to users. So preliminary, only make a k8s cluster upgrade while the
-cluster is unused or that the maintenance is communicated ahead of time.
-```
```
Comment (consideRatio) [inline docs/howto/upgrade-cluster/aws.md:15]: We still haven't, but as long as we are only very little disruptive (brief networking etc) we can probably make upgrades while clusters are active anyhow.

I opted to remove this rather than refine this. I acknowledge we should have a policy, but I'd like us to collectively iterate and find agreement after collective experience rather than me declaring and motivating one in a PR about how to make a k8s upgrade technically.

**Review by GeorgianaElena — APPROVED** [review]

**Comment by consideRatio** [comment]
Thanks for reviewing @GeorgianaElena!! I rebased and added a commit with some adjustments after trying them out myself.
